freqtrade/exchange/__init__.py

```diff
@@ -95,6 +95,10 @@ def _init_ccxt(self, exchange_config: dict) -> ccxt.Exchange:
         except (KeyError, AttributeError):
             raise OperationalException(f'Exchange {name} is not supported')
 
+        # check if config requests sanbox, if so use ['test'] from url
+        if (exchange_config.get('sandbox')):
```

Comment:
What test are you thinking? 
The only one that I can think is check url in ['test'] is not eq to url in ['api']
We're dependent on CCXT being a good custodian of service as for all trades.

Reply:
agree with that, i would not try and test ccxt for correctness - which would be difficult and should not be in this project - but our code (so if i set `"sandbox"`, the sandbox-url is actually used).

i wonder if we should log an error and Raise an (appropriate) exception if sandbox is configured and the exchange does not support sandbox.

Running this now (with Sandbox) against binance will result in a crash with "KeyError" without showing the user any reason what's wrong. (i know because i intentionally tested the expected failure ...)

```
2018-07-28 19:43:28,822 - freqtrade.strategy - DEBUG - Imported strategy unknown.DefaultStrategy as freqtrade.strategy.DefaultStrategy
2018-07-28 19:43:28,822 - freqtrade.exchange - INFO - Instance is running with dry_run enabled
2018-07-28 19:43:28,825 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/main.py", line 45, in main
    freqtrade = FreqtradeBot(config)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 54, in __init__
    self.exchange = Exchange(self.config)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/exchange/__init__.py", line 66, in __init__
    self._api = self._init_ccxt(exchange_config)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/exchange/__init__.py", line 100, in _init_ccxt
    api.urls['api'] = api.urls['test']
KeyError: 'test'

```


so i would want to see 2 tests 
- testing if the url changes
- testing it errors (logs the errormessage) and fails if i use this setting against binance